### ISSUE TITLE
How can I enable asciimath support?

### ISSUE THREAD

**Issue by MarcoIeni** [asciidoctor/asciidoctor-fopub#49]
I have an adoc file with some "asciimath:[formula]".
If I use the command "asciidoctor file.adoc" I get an html file where I can see my formulas correctly, but if I follow the procedure with fopub I can't see my formulas in the pdf.

**Comment by mojavelinux** [comment]
There are two parts. First, you need to install the asciimath gem:

    $ gem install asciimath

With that gem installed, Asciidoctor will automatically convert AsciiMath to MathML when the `stem` attribute is enabled.

    $ asciidoctor -b docbook -a stem sample.adoc

NOTE: Asciidoctor 1.5.3 only converts AsciiMath blocks. Asciidoctor 1.5.4 will also convert inline AsciiMath elements (as in your example). See https://github.com/asciidoctor/asciidoctor/issues/1622.

Next, we need to add the JEuclid plugin for FOP to fopub. I've done that in PR #50 (along with the instruction above). Once this is merged, you just need to do a clean Gradle build and it will become available:

    $ ./gradlew clean installApp

**Comment by MarcoIeni** [comment]
Thank you, but it doesn't helped me.
I followed your instructions and it doesn't worked the first time.
I downloaded the zip of the project and I tried once again but it seems that asciidoctor skips "asciimath:[]"

This is my sample.adoc:

```
= My Diabolical Mathmatical Opus
Jamie Moriarty
:stem:

sample1

asciimath:[sqrt(4) = 2] 

sample2
```

and this is the sample.xml that I get:

```
<?xml version="1.0" encoding="UTF-8"?>
<?asciidoc-toc?>
<?asciidoc-numbered?>
<!DOCTYPE article [<!ENTITY % db5ent PUBLIC "-//FOPUB//ENTITIES Entities for DocBook 5" "db5.ent"> %db5ent;]>
<article xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.0" xml:lang="en">
<info>
<title>My Diabolical Mathmatical Opus</title>
<date>2015-12-20</date>
<author>
<personname>
<firstname>Jamie</firstname>
<surname>Moriarty</surname>
</personname>
</author>
<authorinitials>JM</authorinitials>
</info>
<simpara>sample1</simpara>
<simpara>sqrt(4) = 2</simpara>
<simpara>sample2</simpara>
</article>
```

I verified that asciidoctor is up to date and that asciimath is installed

**Comment by mojavelinux** [comment]
That's because the pull request isn't merged yet. I still need to review it.

In the meantime, simply use a math block.

```
= My Diabolical Mathmatical Opus
Jamie Moriarty
:stem:

sample1

[asciimath]
++++
sqrt(4) = 2
++++

sample2
```

**Comment by mojavelinux** [comment]
To clarify, the pull request in Asciidoctor core that converts inline asciimath to MathML is not merged yet. See https://github.com/asciidoctor/asciidoctor/pull/1623.

**Comment by MarcoIeni** [comment]
ok, sorry, and thank you very math for your amazing support!

**Comment by mojavelinux** [comment]
:+1:

**Comment by Walckiers** [comment]
Can Asciidoctor 1.5.2 convert asciimath to format html?

I copied your example
«In the meantime, simply use a math block.»
into file _more_testing.adoc_

and ran
    `asciidoctor  -a stem more_testing.adoc`

and all I got was this

    sample1
    \$sqrt(4) = 2\$

    sample2

(cut and pasted from the html)

I also tried
    `asciidoctor -b html -a stem  more_testing.adoc`
and
    `asciidoctor more_testing.adoc`
           
always with the same result
Is it something to do with  v1.5.2 ?

Richard H

--
I've already done
  ` sudo apt-get upgrade asciidoctor`

But I seem stuck at v 1.5.2 :-   
   ` asciidoctor -v `
says
    Asciidoctor 1.5.2 [http://asciidoctor.org]
    Runtime Environment (ruby 1.8.7
    (2015-04-14 MBARI 8/0x8770 on patchlevel 375) [i686-linux],
    MBARI 0x8770, Ruby Enterprise Edition 2012.02)

And I've also done
    `sudo gem install asciimath`

**Comment by mojavelinux** [comment]
I should have mentioned in the instructions that you need at least Asciidoctor 1.5.4 for full AsciiMath support in the DocBook output. (I'm going to open an issue in the asciidoctor.org repository to document the AsciiMath support in DocBook via the asciimath gem).

> (cut and pasted from the html) all I got was this
> ```
> sample1
> \$sqrt(4) = 2\$
>
> sample2

Correct. When converting to HTML, math is processed in the browser by MathJax. You won't see any MathML in the HTML itself.

> I've already done
> ```
> $ sudo apt-get upgrade asciidoctor
> ```

The 1.5.4 deb is still propagating. It's now available in "stretch" and "sid" (see https://packages.debian.org/search?searchon=names&keywords=asciidoctor) but not yet available in Ubuntu (see http://packages.ubuntu.com/search?keywords=asciidoctor&searchon=names&suite=wily&section=all).

**Comment by Walckiers** [comment]
Thank you for quick and very helpful reply.
> but not yet available in Ubuntu 

I hope that someone from the Ubuntu powerhouse is listening ...

Richard H

**Comment by mojavelinux** [comment]
You could just update the gem using:

    gem update asciidoctor

given that you've already used the gem command to install asciimath.

If that fails, try:

    gem install asciidoctor

**Comment by Walckiers** [comment]
> gem install asciidoctor

didnt you mean
_**sudo**_  gem install asciidoctor ?

No but seriously, that is magic. Look - 

![61047_asciidoctorscreenshot](https://cloud.githubusercontent.com/assets/3853503/13728132/323b6f62-e90e-11e5-8209-8321429f5351.png)


P.S. I think I saw somewhere that you are re-writing asciidoctor documentation. If you need a lambda user for field tests, pl. let me know (off list).

**Comment by mojavelinux** [comment]
> sudo gem install asciidoctor ?

In your case, yes.

> I think I saw somewhere that you are re-writing asciidoctor documentation.

Yes, we are hard at work on it. See https://github.com/asciidoctor/asciidoctor.org/pulse for activity. (The discussion primarily takes place through issues...all in the open).
